framework/src/meshmodifiers/ElementDeleterBase.C

```diff
@@ -183,7 +183,9 @@ ElementDeleterBase::modify()
         Elem * elem = mesh.elem_ptr(r.first);
         const unsigned int side = r.second;
 
-        elem->set_neighbor(side, libmesh_nullptr);
+        libmesh_assert(elem->neighbor(side) == remote_elem);
```

Comment:
Oh, we use `mooseAssert()` instead of `libmesh_assert()`. Ours requires a message as the second parameter but otherwise wraps libMesh's version.